packages/actor-init-sparql/test/system/Engine-test.ts

```diff
@@ -185,6 +187,26 @@ describe('System test: ActorInitSparql', () => {
             quads.map(q => q.object.value.toUpperCase()),
           );
         });
+
+        it('handles complex queries with groupBy', async() => {
+          const context = <any> { sources: [ store ]};
+          const complexQuery = `PREFIX func: <http://example.org/functions#>
+        SELECT (SUM(func:count-chars(?o)) AS ?sum) WHERE {
+              ?s ?p ?o.
+        }
+          `;
+          context[KeysInitSparql.extensionFunctions] = {
+            async 'http://example.org/functions#count-chars'(args: RDF.Term[]) {
+              const arg = args[0];
```

Comment:
Could you also add a variant of this test that is slower (to check for race-conditions)?

We could do this by adding something like this on the current line: `await new Promise((resolve) => setImmediate(resolve, 1));`

Reply:
Ah, I actually meant `setTimeout`, not `setImmediate`. (my bad...)